pkg/attestation/crafter/runners/oidc/github.go

```diff
@@ -0,0 +1,277 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The code is a modified version of the code from the SLSA GitHub generator
+// https://github.com/slsa-framework/slsa-github-generator.
+
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"slices"
+
+	"bytes"
+	"io"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+)
+
+var defaultActionsProviderURL = "https://token.actions.githubusercontent.com"
+
+const (
+	requestTokenEnvKey = "ACTIONS_ID_TOKEN_REQUEST_TOKEN"
+	requestURLEnvKey   = "ACTIONS_ID_TOKEN_REQUEST_URL"
+)
+
+type GitHubOIDCClient struct {
+	requestURL   *url.URL
+	verifierFunc func(context.Context) (*oidc.IDTokenVerifier, error)
+	bearerToken  string
+	audience     []string
+	token        *GitHubToken
+}
+
+// Token represents the contents of a GitHub OIDC JWT token.
+type GitHubToken struct {
+	oidc.IDToken
+
+	// JobWorkflowRef is a reference to the current job workflow.
+	JobWorkflowRef string `json:"job_workflow_ref"`
+
+	// RunnerEnvironment is the environment the runner is running in.
+	RunnerEnvironment string `json:"runner_environment"`
+
+	// RawToken is the unparsed oidc token.
+	RawToken string
+}
+
+// Option is a functional option for configuring a GitHubOIDCClient.
+type Option func(*GitHubOIDCClient)
+
+// WithAudience sets the audience for the OIDC token.
+func WithAudience(audience []string) Option {
+	return func(c *GitHubOIDCClient) {
+		c.audience = audience
+	}
+}
+
+// NewOIDCGitHubClient returns new GitHub OIDC provider client.
+func NewOIDCGitHubClient(opts ...Option) (*GitHubOIDCClient, error) {
+	var c GitHubOIDCClient
+
+	// Get the request URL and token from env vars
+	requestURL := os.Getenv(requestURLEnvKey)
+	if requestURL == "" {
+		return nil, fmt.Errorf("url: %s environment variable not set; does your workflow have `id-token: write` scope?", requestURLEnvKey)
+	}
+
+	parsedURL, err := url.ParseRequestURI(requestURL)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"%w: invalid request URL %q: %w; does your workflow have `id-token: write` scope?",
+			errURLError,
+			requestURL, err,
+		)
+	}
+
+	bearerToken := os.Getenv(requestTokenEnvKey)
+	if len(bearerToken) == 0 {
+		return nil, fmt.Errorf("token: %s environment variable not set; does your workflow have `id-token: write` scope?", requestTokenEnvKey)
+	}
+
+	c = GitHubOIDCClient{
+		requestURL:  parsedURL,
+		bearerToken: bearerToken,
+	}
+
+	// Apply the options
+	for _, opt := range opts {
+		opt(&c)
+	}
+
+	c.verifierFunc = func(ctx context.Context) (*oidc.IDTokenVerifier, error) {
+		provider, err := oidc.NewProvider(ctx, defaultActionsProviderURL)
+		if err != nil {
+			return nil, err
+		}
+		return provider.Verifier(&oidc.Config{
+			// we skip the check since we are not using a client IDs
+			SkipClientIDCheck: true,
+		}), nil
+	}
+
+	return &c, nil
+}
+
+func (c *GitHubOIDCClient) IsHosted(_ context.Context) bool {
```

Comment:
this should not have the implementation here, 

Basically, this package is just a oidc client